app/controllers/spree/users_controller_decorator.rb

```diff
@@ -15,4 +15,10 @@ def show
 
     @orders = @orders.where('distributor_id != ?', Spree::Config.accounts_distributor_id)
   end
+
+  # Endpoint for queries to check if a user is already registered
+  def registered_email
+    user = Spree.user_class.find_by_email params[:email]
+    render json: { registered: user.present? }
+  end
```

Comment:
I'm missing at least a controller test for this.

Reply:
Spec added.